core/src/marker/markerManager.cpp

```diff
@@ -297,6 +298,27 @@ bool MarkerManager::buildStyling(Marker& marker) {
     try {
         // Update the draw rule for the marker.
         YAML::Node node = YAML::Load(marker.stylingString());
+
+        // Check if marker styling defines a layer in the yaml scene
+        if (Node layer = node["layer"]) {
+            if (!layer.IsScalar()) {
+                LOGE("layer for marker styling needs to be a scalar value.");
+                return false;
+            }
+
+            const DataLayer* sceneLayer = m_scene->findLayer(layer.Scalar());
+            if (sceneLayer && !sceneLayer->rules().empty()) {
+                // select first drawRule from this layer's rules
+                auto rule = sceneLayer->rules().front();
```

Comment:
👍  Will see if I can reuse the rule merging aptly to get this to a less "limiting" zone.

With respect to allowing more than one draw groups to draw a marker, we will have to modify the marker architecture to have meshes for more than one "styles". Can be done, but I would prefer to make that another task and not mingle it with this one.